cty/convert/public_test.go

```diff
@@ -581,6 +591,64 @@ func TestConvert(t *testing.T) {
 			Type: cty.Object(map[string]cty.Type{"foo": cty.String}),
 			Want: cty.ObjectVal(map[string]cty.Value{"foo": cty.StringVal("hello")}),
 		},
+		// reduction of https://github.com/hashicorp/terraform/issues/23804
+		{
+			Value: cty.ObjectVal(map[string]cty.Value{
+				"a": cty.ObjectVal(map[string]cty.Value{
+					"x": cty.TupleVal([]cty.Value{cty.StringVal("foo")}),
+				}),
+				"b": cty.ObjectVal(map[string]cty.Value{
+					"x": cty.TupleVal([]cty.Value{cty.StringVal("bar")}),
+				}),
+				"c": cty.ObjectVal(map[string]cty.Value{
+					"x": cty.TupleVal([]cty.Value{cty.StringVal("foo"), cty.StringVal("bar")}),
+				}),
+			}),
+			Type: cty.Map(cty.Map(cty.DynamicPseudoType)),
+			Want: cty.MapVal(map[string]cty.Value{
+				"a": cty.MapVal(map[string]cty.Value{
+					"x": cty.ListVal([]cty.Value{cty.StringVal("foo")}),
+				}),
+				"b": cty.MapVal(map[string]cty.Value{
+					"x": cty.ListVal([]cty.Value{cty.StringVal("bar")}),
+				}),
+				"c": cty.MapVal(map[string]cty.Value{
+					"x": cty.ListVal([]cty.Value{cty.StringVal("foo"), cty.StringVal("bar")}),
+				}),
+			}),
+		},
+		// reduction of https://github.com/hashicorp/issues/24167
```

Comment:
```suggestion
		// reduction of https://github.com/hashicorp/terraform/issues/24167
```

Possibly my most nitpicky nitpick yet?

Reply:
Good eye! Thank you, fixed